tools/shared.py

```diff
@@ -524,6 +524,33 @@ def check_sanity(force=False):
   EM_CONFIG (so, we re-check sanity when the settings are changed).  We also
   re-check sanity and clear the cache when the version changes"""
   ToolchainProfiler.enter_block('sanity')
+
+  def run_npm_install_if_needed():
```

Comment:
Sorry, I mean "as opposed to a top-level function"? It being scoped locally keeps it from being called elsewhere, but it's long enough that it breaks up the flow of `check_sanity` and makes it even harder to read. It's well-named so it could live out of line and not hurt I think.